File: week-1-c/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)

{

    // Declaring my variables
    int w, x, y, z, a;

    // Prompting user for height between 1 and 8, inclusive
    do
    {
        w = get_int("Height: ");
    } while (w < 1 || w > 8);

    // This loop prints a new row
    for (x = 0; x < w; x++)
    {

        // This loop prints the first set of spaces
        for (y = 0; y < w - x - 1; y++)
        {
            printf(" ");
        }

        // This loop prints the first set of hashes
        for (z = 0; z <= x; z++)
        {
            printf("#");
        }

        // This prints the 2 spaces that separate the 2 pyramids
        printf("  ");

        // This prints the rightmost pyramid
        for (a = 0; a <= x; a++)
        {
            printf("#");
        }
        printf("\n");
    }
}

File: week-6-python/sentimental-mario-more/mario.py
from cs50 import get_int


# TODO


def main():

    # repeat get_int as long as the user doesn't input a number between 1 and 8
    while True:
        height = get_int("Height: ")

        if height >= 1 and height <= 8:
            break

        else:
            print("Height must be an integer between 1 and 8!")

    # draw the pyramid
    draw(height)


def draw(height):

    # iterate through each row and print the spaces and hashes required
    for row in range(height):
        print(' ' * (height - row - 1) + '#' * (row + 1) + '  ' + '#' * (row + 1))


if __name__ == "__main__":
    main()
